### ISSUE TITLE
Installation problems on Windows - The application was unable to start correctly (0xc000007b)

### ISSUE THREAD

**Issue by marcello-pietrobon** [conda-forge/miniforge#229]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

I'm on Windows 7 - 64 bits
using Miniforge3-Windows-x86_64-4.10.3-7.exe

I uninstalled any other version of Python on my desktop.

I get this error during installation,

`Failed to initialize anaconda directory`

and if I ignore it, like it was suggested here
[Anaconda3 2.4 with python 3.5 installation error (procedure entry not found; Windows 10) - Stack Overflow](https://stackoverflow.com/questions/33625683/anaconda3-2-4-with-python-3-5-installation-error-procedure-entry-not-found-win)

 the installation becomes successful.
But then when I run python, or other executables in the miniforge3 directory, I get the message

`The application was unable to start correctly (0xc000007b). Click OK to close the application. `

My guess is that some MS redistributable are missing, but which one? My guess is VS-2019. Can you confirm?

Are there other possible reasons for this error?

Thank you,

**Comment by jakirkham** [comment]
Sounds like there is an existing installation, which is causing issues. Have you tried removing existing installs and starting from scratch?

**Comment by hmaarrfk** [comment]
I think it is also possible that you have a space in your username. I wasn't able to install with a space in mine a while ago.

**Comment by marcello-pietrobon** [comment]
Thank you. for the hint.

I write few notes here, just for the record, but unfortunately I had to give up and I'll switch to a Linux distro when using Julia with Python.

The following notes are about Anaconda, both because by analyzing `miniforge` the issue I experience appears pretty much identical (in different forms it's always the error `0xc000007b`  causing this), and because I used `PythonCall` in Julia which did let me  to better isolate the issue.
[PyCall No system-wide Python was found](https://discourse.julialang.org/t/pycall-no-system-wide-python-was-found/71605/5)

I changed username, without space, and things improved in the sense that with `Python3.8 (Anaconda3-2021.05)` things worked better for a moment and I was actually able to see a plot. Just once. With `Python3.9 (Anaconda3-2021.11` things didn't work at all for me.

I analyzed the problem with the `DependencyWalker` tool. It doesn't show any significant error for what concerns `Python 3.8`, while with `Python3.9` there's a few. Maybe they are all of them related to the `API-MS-WIN-CORE-PATH-L1-1-0.DLL`, because it's a 32-bit dll and it *does* contain code, not just resources. At the same time there's no 64 bits versions around, so I really wonder, it doesn't make much sense to internally use IPC in order to communicate with a DLL that is only having functions like PathCchCanonicalize and so on, but I don't know.
I especially wonder if really nobody experiences any error of this kind under Windows7.

Best,
M

**Comment by bramson** [comment]
Yes, so this issue persists.  I couldn't find any specific claim that Windows 7 is not supported, but I get this exact same behavior.  

Running the installer produces:
`Failed to initialize anaconda directory`
and if I ignore it, the installation becomes successful...
but then when I run python, or other executables in the miniforge3 directory, I get the message
"The program can't start because api-ms-win-core-path-l1-1-0.dll is missing from your computer.  Try reinstalling the program to fix this problem"
Of course that doesn't fix anything.  

So I updated the C++ redistributable from Microsoft, nothing changed.

So I then added that specific dll file specifically.

I remove and try to reinstall miniforge and I still get the `Failed to initialize Anaconda directories` message.
And now when I run it, I get `The application was unable to start correctly (oxc000007b). Click OK to close the application.`

So it's a complete failure and waste of 2 hours.
I'll try WinPython and/or other distributions, but it's a shame that this error persists, and without even any serious comments or suggestions to fix it.

**Comment by hmaarrfk** [comment]
I'm sorry, but windows 7 is quite old and I don't think anybody has plans to help support it. I will move this issue to a discussion in case you want to help document your adventure, but if you are trying to get work done, I suggest updating to windows 10 (paid option) or installing Ubuntu (free option)
